### DISCUSSION TITLE
Libre 2 NFC scan

### DISCUSSION THREAD

**Discussion by adenis76** [NightscoutFoundation/xDrip#3561] [General]
Hi,

Currently xdrip can't scan libre 2 with Nfc and also not retrieve history from patched librelink app. Please add Nfc scan for libre 2 and possibility to retrieve missed reading.

Thank you.

**Comment by MasterPlexus** [comment]
@adenis76, unfortunately there is at the moment no way to add this functionality to xdrip. In Libre 2 everything is encrypted, and the keys are not known. If nobody can get the keys, a direct scan will not work.

**Comment by nsghub** [comment]
There is a new Blucon Nightrider for the Libre 2. So somebody already found the keys. Maybe someone could talk to the guys from Ambrosia.

**Comment by chaosbiber** [comment]
Ambrosia wasn't the first. There was a group that found a first solution around April(?) this year and offered Ambrosia, MiaoMiao and alike the libraries if they make a donation to the Nightscout foundation. Don't know if any company uses that lib or if they did something on their own. But the problem is: those readings aren't as accurate as the data that comes from the patched libre app. They'll differ quite a bit and are much more noisy. So the question is: is it worth the effort of implementing that function in xDrip until the protocols of libre2 are fully reverse engineered?

**Comment by adenis76** [comment]
My real "problem" here is gap on xdrip and nightscout graph when phone lose connection with sensor. With blucon and libre 1 sensor xdrip retrieve missed libre history. Can we add possibility to xdrip to retrieve missed readings from patched librelink app?

**Comment by mysticprog** [comment]
So, Libre 2 over Bluecon is not supported ? Is MiaoMiao 2 supported i xdrip ?
Thanks.

**Comment by tzachi-dar** [comment]
We are working on this.

**Comment by mysticprog** [comment]
Ok. Thanks

**Comment by dideldum** [comment]
The patched LibreLink sends also an Intent "com.librelink.app.ThirdPartyIntegration.SENSOR_SCAN" with all the scanned values (historic and realtime). I tried it in a small test app and all the scanned values seem to be there.
It's just xDrip+, which doesn't take this intent and fills the gap with the scanned values.
Couldn't xDrip+ implement the reception of this patched LibreLink intent?

**Comment by mysticprog** [comment]
> The patched LibreLink sends also an Intent "com.librelink.app.ThirdPartyIntegration.SENSOR_SCAN" with all the scanned values (historic and realtime). I tried it in a small test app and all the scanned values seem to be there.
> It's just xDrip+, which doesn't take this intent and fills the gap with the scanned values.
> Couldn't xDrip+ implement the reception of this patched LibreLink intent?

I'm just interested. Why do you need it? Well you have readings in XDrip every 5 minutes so I don't see the point in the suggestion.

**Comment by dideldum** [comment]
> I'm just interested. Why do you need it? Well you have readings in XDrip every 5 minutes so I don't see the point in the suggestion.

We even get readings from LibreLink in xDrip+ every minute (not just every 5), so this is great. But if you leave the device with the patched LibreLink/xDrip+ somewhere and miss values, you cannot fill the gap currently in xDrip+.

With transmitters like MiaoMiao this is done, the gap is filled. With the patched LibreLink this is also possible, xDrip+ just must receive the intent and handle the scanned data.

It's just the matter of filling gaps!

**Comment by Navid200** [comment]
What is the update on this?

**Comment by Navid200** [comment]
Please update if there is still no work-around for this and reopen.

**Comment by tarlepp** [comment]
I'm having this issue ATM - I need to scan with libre so that I can provide that information to my health center. Is there some setting in xdrip that I need to change to allow also use libre reader + xdrip together?

**Comment by Navid200** [comment]
@psonnera @tzachi-dar 
Is this still an issue?

**Comment by psonnera** [comment]
@Navid200 this is not an issue with Libre 2 EU that is supported as well as Libre 1, 1 US and Pro with NFC scan in xDrip+ (might require OOP2) https://xdrip.readthedocs.io/en/latest/install/libreNFC/#supported-sensors

**Comment by tarlepp** [comment]
I have Libre 2 EU sensor and it's working just fine with xDrip+ but the problem is that IF I use xDrip+ with my miaomiao reader I cannot use Libre reader OR Libre reader app in my phone - those cannot find the sensor at all.

When I first started to use xDrip+ Libre and phone app worked as expected, but after that two last sensors I'm seeing this behaviour - So I guess that "something" has changed on the sensor itself etc.

**Comment by psonnera** [comment]
@tarlepp if you're using a Miaomiao2 that's a known issue. Try to disable xDrip+ and check the issue remains, this would confirm it's bridge relevant and not xDrip+. Next sensor check the angle at which you installed it and try different rotation to reduce the shielding effect on the sensor antenna.

**Comment by tarlepp** [comment]
I'm using MiaoMiao3 and not MiaoMiao2, so most likely some another issue.

**Comment by jain-shreyansh** [comment]
> We are working on this.

any progress?
